src/js/signin/containers/Main.jsx

```diff
@@ -0,0 +1,173 @@
+      method: 'GET',
+    }).then(response => {
+      return response.json();
+    }).then(json => {
+      this.props.onUpdateVerifyUrl(json.identity_proof_url);
+    });
+  }
+
+  setMyToken(event) {
+    if (event.data === sessionStorage.userToken) {
+      this.getUserData(this.props.dispatch);
+      this.getLogoutUrl();
+    }
+  }
+
+  getLogoutUrl() {
+    this.logoutUrlRequest = fetch(`${environment.API_URL}/v0/sessions`, {
+      method: 'DELETE',
+      headers: new Headers({
+        Authorization: `Token token=${sessionStorage.userToken}`
+      })
+    }).then(response => {
+      return response.json();
+    }).then(json => {
+      this.props.onUpdateLogoutUrl(json.logout_via_get);
+    });
+  }
+
+  handleLogin(loginUrl = 'idme') {
+    this.loginUrlRequest = handleLogin(this.props.login.loginUrls[loginUrl], this.props.onUpdateLoginUrl);
+  }
+
+  handleSignup() {
+    window.dataLayer.push({ event: 'register-link-clicked' });
+    const myLoginUrl = this.props.login.loginUrls.idmeUrl;
+    if (myLoginUrl) {
+      window.dataLayer.push({ event: 'register-link-opened' });
+      const receiver = window.open(`${myLoginUrl}&op=signup`, '_blank', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
+      receiver.focus();
+    }
+  }
+
+  handleLogout() {
+    window.dataLayer.push({ event: 'logout-link-clicked' });
+    const myLogoutUrl = this.props.login.logoutUrl;
+    if (myLogoutUrl) {
+      window.dataLayer.push({ event: 'logout-link-opened' });
+      const receiver = window.open(myLogoutUrl, '_blank', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
+      receiver.focus();
+    }
+  }
+
+  checkTokenStatus() {
+    if (sessionStorage.userToken) {
+      if (moment() > moment(sessionStorage.entryTime).add(45, 'm')) {
+        // TODO(crew): make more customized prompt.
+        if (confirm('For security, you’ll be automatically signed out in 2 minutes. To stay signed in, click OK.')) {
+          this.handleLogin();
+        } else {
+          this.handleLogout();
+        }
+      } else {
+        if (this.getUserData(this.props.dispatch)) {
+          this.props.onUpdateLoggedInStatus(true);
+        }
+      }
+    } else {
+      this.props.onUpdateLoggedInStatus(false);
+    }
+  }
+
+  render() {
+    const currentlyLoggedIn = this.props.login.currentlyLoggedIn;
+
+    if (this.props.verify && currentlyLoggedIn) {
+      return (
+        <Verify
+          verifyUrl={this.props.login.verifyUrl}/>
+      );
+    }
+
+    return (
+      <Signin
+        currentlyLoggedIn={currentlyLoggedIn}
+        handleSignup={this.handleSignup}
+        handleLogin={this.handleLogin}/>
+    );
+  }
+}
+
+const mapStateToProps = (state) => {
+  const userState = state.user;
+  return {
+    login: userState.login,
+    profile: userState.profile
+  };
+};
+
+
+const mapDispatchToProps = (dispatch) => {
+  return {
```

Comment:
Any reason to not use the object syntax for mapDispatchToProps?

Reply:
A lot of this logic is copied over from the `login` app, which heavily used `mapDispatchToProps` in this style. We're going to be refactoring the code after the initial launch, and the object syntax for `mapDispatchToProps` is on the plate.